src/librustc/lint/builtin.rs

```diff
@@ -346,6 +346,12 @@ declare_lint! {
     "outlives requirements can be inferred"
 }
 
+declare_lint! {
+    pub DUPLICATE_MATCHER_BINDING_NAME,
```

Comment:
Nit: [Lint naming conventions](https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#lints).
`duplicate_matcher_binding_name` -> `duplicate_matcher_binding_names`
(Not too important for soon-to-be-removed deprecation warnings but still.)